File: app/src/main/java/com/jefferson/application/br/activity/triggers/SwitchVisibilityTrigger.java
package com.jefferson.application.br.activity.triggers;

import android.view.View;

public class SwitchVisibilityTrigger {

    private final View view;

    public SwitchVisibilityTrigger (View view) {
       this.view = view;
    }

    public void switchVisibility() {
        int visibility = getVisibility() == View.VISIBLE? View.GONE:View.VISIBLE;
        view.setVisibility(visibility);
    }

    public int getVisibility() {
        return view.getVisibility();
    }
}

File: app/src/main/java/com/jefferson/application/br/triggers/SwitchVisibilityTrigger.kt
package com.jefferson.application.br.triggers

import android.view.View

class SwitchVisibilityTrigger(private val view: View) {
    fun switchVisibility() {
        val visibility = if (visibility == View.VISIBLE) View.GONE else View.VISIBLE
        view.visibility = visibility
    }

    val visibility: Int
        get() = view.visibility
}